common/src/main/scala/uk/ac/wellcome/platform/stacks/common/services/CatalogueService.scala

```diff
@@ -74,16 +71,23 @@ class CatalogueService(
       // Ensure we are only matching items that match the passed id!
       filteredItems = identifier match {
         case CatalogueItemIdentifier(id) =>
-          items.filter(_.id.catalogueId.value == id)
+          items.filter(_.catalogueId.value == id)
         case SierraItemIdentifier(id) =>
-          items.filter(_.id.sierraId.value == id)
+          items.filter(_.sierraId.value == id)
         case StacksWorkIdentifier(id) =>
-          items.filter(_.id.catalogueId.value == id)
+          items.filter(_.catalogueId.value == id)
         case _ => items
       }
 
-    } yield filteredItems match {
-      case List(item) => Some(item)
-      case _          => None
-    }
+      distinctFilteredItems = filteredItems.distinct
```

Comment:
Worth a comment about why it's okay to `.distinct` here (we're looking for unique id->id mappings, and the same item might be on multiple works).